ansible/roles-infra/infra-osp-template-create/tasks/create_osp_template.yml

```diff
@@ -2,86 +2,71 @@
 - name: Wait for less than {{ osp_max_create_in_progress }} Heat stacks in CREATE_IN_PROGRESS
   command: >-
     openstack stack list --property stack_status=CREATE_IN_PROGRESS --format json
-  environment:
-    OS_AUTH_URL: "{{ osp_auth_url }}"
-    OS_USERNAME: "{{ osp_auth_username | default(osp_auth_username_member) }}"
-    OS_PASSWORD: "{{ osp_auth_password | default(osp_auth_password_member) }}"
-    OS_PROJECT_NAME: "{{ osp_project_name }}"
-    OS_PROJECT_DOMAIN_ID: "{{ osp_auth_project_domain }}"
-    OS_USER_DOMAIN_NAME: "{{ osp_auth_user_domain }}"
+  environment: "{{ __infra_osp_template_create_environment }}"
   register: r_stack_list
-  until: 
+  until:
   - r_stack_list.stdout | from_json | count <= osp_max_create_in_progress
   retries: 10
   delay: "{{ osp_max_create_in_progress_base_delay | random(start=osp_max_create_in_progress_random_delay) }}"
+  changed_when: false
 
 - name: Show the number of creating stacks
   debug:
     msg: The number of creating stacks is {{ r_stack_list.stdout | from_json | count }}
 
-- environment:
-    OS_AUTH_URL: "{{ osp_auth_url }}"
-    OS_USERNAME: "{{ osp_auth_username_member }}"
-    OS_PASSWORD: "{{ osp_auth_password_member }}"
-    OS_PROJECT_NAME: "{{ osp_project_name }}"
-    OS_PROJECT_DOMAIN_ID: "{{ osp_auth_project_domain }}"
-    OS_USER_DOMAIN_NAME: "{{ osp_auth_user_domain }}"
+- name: Check if heat stack already exists
+  command: >-
+    openstack stack show -f json base-stack-{{ guid }}
+  environment: "{{ __infra_osp_template_create_member_environment }}"
+  register: r_stack_show
+  changed_when: false
+  failed_when: false
+  tags: heat
+
+- when: >-
+    __stack_status in ['CREATE_FAILED', 'NONE']
+  vars:
+    __stack: "{{ r_stack_show.stdout | from_json if r_stack_show.rc == 0 else {} }}"
```

Comment:
Can we not use double underscores, please? How about r_ instead like everywhere else.

Reply:
The `r_` prefix is used for registered variables. This is just a task level variable.

The use of underscore is to confirm to the standards set by the roles standards working group, documented here:
https://github.com/oasis-roles/meta_standards, "internal variables (those that are not expected to be set by users) are to be prefixed by two underscores"

Reply:
I would normally go with a single underscore, but the working group went with two.

Reply:
Did we discussed about following the standards working group guidelines and rules ?
(i'm open, just i'm not sure we decided on that for agnosticd).

Reply:
I don't believe we did. @tonykay and I participated in some of the roles standards development. The standards documented for oasis are fairly new and are being raised to more visibility within Red Hat. Their rules are quite close to our own. They don't have an `r_` prefix standard for registered variables... but I thought we were already on the same page with private vars starting with underscore?

Reply:
Why not follow the usual conventions and also prefix private variables with the config / workload name?

Reply:
The `__stack` and `__stack_status` vars are declared at the task level and are not defined outside of the task, so I felt a short name was more readable and helps make it clear that these are scoped to the task.

In contrast, `__infra_osp_template_create_retry_count` and `__infra_osp_template_create_environment` and `__infra_osp_template_create_member_environment` are exposed outside of the role because of how they are declared and so those include the role name.